extensions/elytron-security-ldap/runtime/src/main/java/io/quarkus/elytron/security/ldap/LdapRecorder.java

```diff
@@ -47,8 +47,8 @@
     private ExceptionSupplier<DirContext, NamingException> createDirContextSupplier(DirContextConfig dirContext) {
         DirContextFactory dirContextFactory = new QuarkusDirContextFactory(
                 dirContext.url,
-                dirContext.principal,
-                dirContext.password);
+                dirContext.principal.orElse(null),
```

Comment:
I'm not 100% sure about the implications of allowing the anonymous LDAP binds OOB. I've looked around, for example, https://www.acunetix.com/vulnerabilities/web/ldap-anonymous-binds/. I guess it should be OK in some cases, but may be we should have a boolean `allow-anonymous-bind` (false by default) ? This will let the empty principal/password will be accepted only if the user has explicitly requested it by having `allow-anonymous-bind=true` ? Without it, with `Optional` we don't know if it is a user mistake or an intentional decision...

Reply:
I'm really not sure about this. I think it's OK as is tbh. If they can't log in to their LDAP directory, they will soon realize that they need to set the credentials.

Reply:
Anonymous binds enabled in LDAP servers are usually seen as a security flaw, but having a flag to allow it seems to be too much work compared to just not providing the credentials, so I'm torn on this one too :)

I think we should at the very least put something in the logs.

Reply:
Hi @gsmet @gastaldi yeah, I agree, it does feel like it is one more extra step; my concern would be that say the next day after the release someone files the vulnerability request along these lines "Quarkus activates the anonymous LDAP binds by default" with some description why it would be a risk :-). As it stands right now, the user who has to get it bound to the secured LDAP has to do some real work (set the principal and password, 2 props), while the user who is OK with the anonymous one just does nothing, so may be it is not that much work to set this property...So at least in the hypothetical scenario (which a good number of CVEs operate in), we have a Quarkus doing an anonymous bind to the LDAP which has been misconfigured to allow such binds. I think I'm getting paranoid, sorry :-)

Reply:
@gsmet @gastaldi If it is not convincing, then go ahead please, I'd not like my theoretical musings stop the enhancement going ahead :-). Would also be good for @stuartwdouglas to check as well but he is offline right now

Reply:
@sberyozkin that makes sense, but isn't up to the LDAP server to decide if anonymous bindings are allowed?

Reply:
@gastaldi yes, makes sense